ftest/src/test/java/org/arquillian/reporter/GreeterReportVerifier.java

```diff
@@ -0,0 +1,229 @@
+
+    private static File reportInJson;
+    private static ExecutionReport executionReport;
+
+    @BeforeClass
+    public static void setUp() throws FileNotFoundException {
+        JUnitCore.runClasses(GreeterTest.class);
+        reportInJson = new File("target/report.json");
+        executionReport = ReportJsonParser.parse("target/report.json");
+    }
+
+    @Test
+    public void verify_report_in_json_exists() throws FileNotFoundException {
+        assertThat(reportInJson).exists();
+    }
+
+    @Test
+    public void verify_execution_report_exists_with_contents() {
+        assertThatExecutionReport(executionReport)
+            .hasName("execution")
+            .hasNumberOfEntries(0)
+            .hasNumberOfSubReports(0);
+    }
+
+    @Test
+    public void verify_test_suite_report_exists_with_contents() {
+        List<TestSuiteReport> testSuiteReports = executionReport.getTestSuiteReports();
+        TestSuiteReport testSuiteReport = testSuiteReports.get(0);
+
+        assertThat(testSuiteReports).size().isEqualTo(1);
+
+        assertThatTestSuiteReport(testSuiteReport)
+            .hasName(ArquillianCoreKey.TEST_SUITE_NAME)
+            .hasNumberOfSubReportsAndEntries(0);
+
+        assertThat(testSuiteReport.getConfiguration().getName())
+            .isEqualTo(ReporterCoreKey.GENERAL_TEST_SUITE_CONFIGURATION_REPORT);
+
+        assertThat(testSuiteReport.getTestClassReports()).size().isEqualTo(1);
+    }
+
+    @Test
+    public void verify_test_suite_configuration_report_exists_with_contents() {
+        ConfigurationReport configuration = executionReport.getTestSuiteReports().get(0).getConfiguration();
+        List<Report> subReports = configuration.getSubReports();
+
+        assertThatReport(configuration)
+            .hasName(ReporterCoreKey.GENERAL_TEST_SUITE_CONFIGURATION_REPORT)
+            .hasNumberOfEntries(0)
+            .hasNumberOfSubReports(2);
+
+        for (Report subReport : subReports) {
+            if (isContainerConfigurationSubReport(subReport)) {
+                verify_container_configuration_sub_report(subReport);
+            }
+            if (isEnvironmentConfigurationSubReport(subReport)) {
+                verify_environment_configuration_sub_report(subReport);
+            }
+        }
+    }
+
+    private void verify_environment_configuration_sub_report(Report subReport) {
+        assertThat(subReport.getName())
+            .isEqualTo(EnvironmentKey.ENVIRONMENT_SECTION_NAME);
+        assertThat(subReport.getEntries()).size().isEqualTo(8);
+        assertThatReport(subReport)
+            .hasEntryContainingKeys(EnvironmentKey.JAVA_VERSION,
+                EnvironmentKey.TEST_RUNNER,
+                EnvironmentKey.TIMEZONE,
+                EnvironmentKey.CHARSET,
+                EnvironmentKey.DOCKER,
+                EnvironmentKey.OPERATIVE_SYSTEM,
+                EnvironmentKey.OPERATIVE_SYSTEM_ARCH,
+                EnvironmentKey.OPERATIVE_SYSTEM_VERSION);
```

Comment:
> Hard code it's values - which won't work in other's machine/CI

No, this is not what I meant.

> Verify it using same logic which is used in creating report using System.getEnv() - this won't fail in any case.
>  So I think we should be interested only in structure or values which are not going to change in any case.

Don't forget that you don't verify only the content itself, but also if the content is there at all. If there isn't any bug in creating section trees or merging logic that would rewrite the values. 
Another benefit is that you verify also the parser logic - that it parses all the strings correctly. Early or later, the parser will be probably moved to separated module to provide a functionality of parsing the report to other services.